src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,267 @@
+            Predicates.equal(INTEGRATION_STATUS, IntegrationStatus.IN_PROGRESS),
+            Predicates.equal(INTEGRATION_STATUS, IntegrationStatus.QUEUED)
+    )
+
+    companion object {
+        @JvmStatic
+        fun init(blackbox: Blackbox, dataModelService: EdmManager) {
+            if (blackbox.enabled) {
+                logEntityType = dataModelService.getEntityType(FullQualifiedName(blackbox.entityTypeFqn))
+            }
+        }
+    }
+
+    init {
+        executor.execute {
+            semaphore.acquire()
+
+            //load any jobs that were in progress or queued
+            integrationJobs.entrySet(statusPredicate).forEach{
+                integrationQueue.put(it.key to it.value.integrationName)
+            }
+
+            while (true) {
+                if (semaphore.availablePermits() > 0) {
+                    loadCargo()
+                }
+            }
+        }
+    }
+
+    fun enqueueIntegrationJob(integrationName: String): UUID {
+        checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist")
+        val jobId = generateIntegrationJobId()
+        integrationQueue.put(Pair(jobId, integrationName))
+        integrationJobs[jobId] = IntegrationJob(integrationName, IntegrationStatus.QUEUED)
+        return jobId
+    }
+
+    private final fun loadCargo() {
+        val integrationJob = integrationQueue.take()
+        val jobId = integrationJob.first
+        val integration = integrations.getValue(integrationJob.second)
+        //val token = MissionControl.getIdToken(creds.getProperty("email"), creds.getProperty("password"))
+        val apiClient = ApiClient(integration.environment) { "testingtoken" }
+        val dataIntegrationApi = apiClient.dataIntegrationApi
+
+        val flightPlan = mutableMapOf<Flight, Payload>()
+        val tableColsToPrint = mutableMapOf<Flight, List<String>>()
+        integration.flightPlanParameters.values.forEach {
+            val srcDataSource = getSrcDataSource(it.source)
+            val payload = JdbcPayload(readRateLimit.toDouble(), srcDataSource, it.sql, fetchSize, readRateLimit != 0)
+            flightPlan[it.flight!!] = payload
+            tableColsToPrint[it.flight!!] = it.sourcePrimaryKeyColumns
+        }
+        val destinationsMap = generateDestinationsMap(integration, missionParameters, dataIntegrationApi)
+
+        val shuttle = Shuttle(
+                integration.environment,
+                flightPlan,
+                entitySets.values.associateBy { it.name },
+                entityTypes.values.associateBy { it.id },
+                propertyTypes.values.associateBy { it.type },
+                destinationsMap,
+                dataIntegrationApi,
+                tableColsToPrint,
+                missionParameters,
+                StorageDestination.S3,
+                blackbox,
+                Optional.of(entitySets.getValue(integration.logEntitySetId.get())),
```

Comment:
I think it might be worth documenting the expectations about the integration object -- i.e. logEntitySetId will be present, it.flight will be non-null (from line 124 above)